game/engine/main.cpp

```
@@ -637,14 +637,29 @@ int minimax(int depth, int alpha, int beta, bool maximizing) {
     vector<Move> moves = generateMoves(side);
+    if (depth > 3 && moves.size() > 25)
+    {
+        depth = 3;
+    }
     orderMoves(moves);
 
     // Filter out moves that leave own king in check
     vector<Move> legal;
+    int checkedMoves = 0;
     legal.reserve(moves.size());
+    int checkedMoves = 0;
```

Clang (14.0.6)

[warning] variable 'checkedMoves' is not initialized

(cppcoreguidelines-init-variables)